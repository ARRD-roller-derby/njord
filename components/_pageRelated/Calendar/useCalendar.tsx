import { useState, useEffect, useContext } from 'react'
import { EventInterface } from '../../../types/Event.interface'
import useFetch from '../../_hooks/useFetch'
import { SocketContext } from '../../../stores/socket.store';

export default function useCalendar() {
  const [between, setBetween] = useState<[Date, Date]>(),
    { data: events, refetch } = useFetch<Array<EventInterface>>('events/events', { between }),
    [triggerRefresh] = useContext(SocketContext);

  useEffect(() => {
    if (between) refetch()
  }, [between])

  useEffect(() => {
    if (triggerRefresh && triggerRefresh?.type === 'event') {
      refetch()
    }
  }, [triggerRefresh])

  return {
    refetch, events: events?.sort((a: any, b: any) => {
      const
        hourA = parseInt(a.hourStart?.slice(0, 2) ?? 0),
        hourB = parseInt(b.hourStart?.slice(0, 2) ?? 0)
      return hourA - hourB
    }), setBetween
  }
}
