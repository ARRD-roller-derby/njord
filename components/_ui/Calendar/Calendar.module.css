.container {
  display: grid;
  grid-template-rows: auto 1fr auto;
  height: 100%;
  align-items: flex-start;
  gap: 20px;
}

.title {
  text-align: center;
}

.calendar {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  gap: 3px;
}

.header {
  text-align: center;
  font-weight: bold;
  text-transform: uppercase;
}

.calendar[data-ismobile='true'] .header {
  font-size: 12px;
}

.buttons {
  display: flex;
  justify-content: space-between;
  gap: 5px;
}

.day {
  display: flex;
  height: 100%;
  justify-content: center;
  cursor: pointer;
}

.day[data-currentmonth='false'] {
  opacity: 0.5;
}

.calendar[data-ismobile='false'] .day {
  background: var(--color-bg-dark);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-height: 80px;
}

.calendar[data-ismobile='false'] .dayNum {
  background: black;
  font-size: 12px;
  padding: 3px;
  text-align: right;
}

.calendar[data-ismobile='false'] .dayNum[data-today='true'],
.calendar[data-ismobile='true'] .dayNum[data-today='true']  {
  background: var(--color-arrd);
  color: white;
}


/** --- MOBILE --- */

.calendar[data-ismobile='true'] {
  gap: 10px;
}

.calendar[data-ismobile='true'] .dayNum {
  background: var(--color-second);
  position: relative;
  font-size: 15px;
  padding: 3px;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}
