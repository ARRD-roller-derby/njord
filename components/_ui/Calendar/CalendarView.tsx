import classes from './Calendar.module.css'
import { CalDayInterface } from '../../../types/CalDay.interface'
import CalendarDay from '../CalendarDay/CalendarDay'
import { EventInterface } from '../../../types/Event.interface'
import EventCreateForm from '../../Events/EventCreateForm/EventCreateForm'
import dayjs from 'dayjs';

interface props {
  readonly cal: Array<any>
  readonly currentMonth: string
  readonly currentMonthNum: number
  readonly nextMonth: Function
  readonly previousMonth: Function
  readonly isMobile: boolean
  readonly isAdmin: boolean
  readonly setPopin: Function
  readonly refetch: Function
  readonly popin: string | null
}

export default function CalendarView({
  cal,
  currentMonthNum,
  nextMonth,
  currentMonth,
  previousMonth,
  isMobile,
  isAdmin,
  setPopin,
  refetch,
  popin,
}: props) {
  return (
    <div className={classes.container}>
      {popin && (
        <EventCreateForm defaultDate={popin} onClose={() => setPopin(null)} refetch={refetch}/>
      )}
      <h1 className={classes.title}>{currentMonth}</h1>
      <div className={classes.calendar} data-ismobile={isMobile}>
        {[
          'lundi',
          'mardi',
          'mercredi',
          'jeudi',
          'vendredi',
          'samedi',
          'dimanche',
        ].map((day) => (
          <div className={classes.header} key={day}>
            {isMobile ? day.slice(0, 3) + '.' : day}
          </div>
        ))}
        {cal.map((day: CalDayInterface) => (
          <CalendarDay
            day={day}
            isAdmin={isAdmin}
            setPopin={setPopin}
            currentMonthNum={currentMonthNum}
            key={day.date.toString()}
          />
        ))}
      </div>
      <div className={classes.buttons}>
        <button onClick={() => previousMonth()}>pr√©cedent</button>
        {isAdmin && <button onClick={() => setPopin(new Date())}>+</button>}
        <button onClick={() => nextMonth()}>suivant</button>
      </div>
    </div>
  )
}
