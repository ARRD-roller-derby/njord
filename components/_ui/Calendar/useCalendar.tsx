import { useEffect, useState } from 'react'
import dayjs from 'dayjs'
import useIsMobile from '../../_hooks/useIsMobile'
import useLocalState from '../../_hooks/useLocalState'
import { CalDayInterface } from '../../../types/CalDay.interface'
import { useSession } from 'next-auth/react';

export default function useCalendar() {
  const 
    {data:session}= useSession(),
    { localState, setLocalState } = useLocalState<{
      year: number
      month: any
    }>(
      {
        year: 0,
        month: dayjs(dayjs().month()),
      },
      'njord_yearMonth'
    ),
    [cal, setCal] = useState<
      Array<CalDayInterface>
    >([]),
    [currentMonth, setCurrentMonth] = useState<string>(''),
    isMobile = useIsMobile()

  function createCalendar() {
    const 
      thisMonth = dayjs().month(localState.month),
      firstDay = dayjs(thisMonth).add(localState.year, 'year').startOf('month'),
      lastDay = dayjs(thisMonth).add(localState.year, 'year').endOf('month'),
      firstCalDay = firstDay.subtract(
        firstDay.day() === 0 ? 0 : firstDay.day() - 1,
        'day'
      ),
      lastCalDay = lastDay.add(
        lastDay.day() === 0 ? 0 : 7- lastDay.day(),
        'day'
      ),
      numOfDay = lastCalDay.diff(firstCalDay, 'day'),
      generateCal = []


    for (let i = 0; i < numOfDay + 1; i++) {
      const day = firstCalDay.add(i, 'day')
      generateCal.push({
        date: day,
        month: day.month(),
        day: day.day(),
        events: [],
      })
    }
    setCurrentMonth(
      dayjs()
        .month(localState.month)
        .add(localState.year, 'year')
        .format('MMMM YYYY')
    )
    //sometimes, one day add
    setCal(generateCal.slice(0,35))
  }

  function nextMonth() {
    const next = localState.month + 1
    if (next > 11) {
      setLocalState({
        month: 0,
        year: localState.year + 1,
      })
    } else {
      setLocalState({
        ...localState,
        month: next,
      })
    }
  }

  function previousMonth() {
    const previous = localState.month - 1

    if (previous < 0) {
      setLocalState({
        month: 11,
        year: localState.year - 1,
      })
    } else {
      setLocalState({
        ...localState,
        month: previous,
      })
    }
  }

  useEffect(() => {
    if (localState) {
      createCalendar()
    }
  }, [localState])

  return {
    cal,
    currentMonthNum: localState?.month,
    nextMonth,
    previousMonth,
    currentMonth,
    isMobile,
    isAdmin:session?.user?.profiles.find((profile:string)=> profile.match(/bureau|coach|com|fest/))
  }
}
