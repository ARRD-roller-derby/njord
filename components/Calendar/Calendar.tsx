import { EventInterface } from '../../types/Event.interface'
import CalendarView from './CalendarView'
import useCalendar from './useCalendar'

interface Props {
  readonly refetch: Function
  readonly events: Array<EventInterface>
  readonly setBetween: Function 
}

export default function Calendar(props:Props) {
  const useProps = useCalendar(props)

  return <CalendarView {...props} {...useProps} />
}
