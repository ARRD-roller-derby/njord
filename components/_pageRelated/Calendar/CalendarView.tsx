import AuthentificatedLayout from '../../_layouts/Authentificated/Authentificated';
import Calendar from '../../Calendar/Calendar';
import useCalendar from './useCalendar';

export default function CalendarView(){
  const props = useCalendar();
  return <AuthentificatedLayout>
    <Calendar {...props}/>
  </AuthentificatedLayout>
}