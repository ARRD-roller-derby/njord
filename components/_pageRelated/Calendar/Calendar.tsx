import CalendarView from "./CalendarView";

export default function Calendar(){
  return <CalendarView/>
}