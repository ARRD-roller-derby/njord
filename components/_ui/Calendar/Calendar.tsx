import CalendarView from './CalendarView'
import useCalendar from './useCalendar'
export default function Calendar() {
  const useProps = useCalendar()

  return <CalendarView {...useProps} />
}
