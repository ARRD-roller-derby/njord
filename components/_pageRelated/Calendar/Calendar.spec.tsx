import { afterEach,describe,expect, it, vi } from 'vitest'
import { cleanup } from '@testing-library/react'
import Calendar from './Calendar'
import { PusherContext } from '../../../stores/pusher.store'
import { render, screen } from '../../../utils/test-utils'
import { rest } from 'msw'
import { server } from '../../../setupFiles/server'

describe('<Calendar />', () => {
  afterEach(cleanup)
  
  server.use(
    rest.post('/api/events/events', (_req, res, ctx) => {
      return res(ctx.status(200), ctx.json([]))
    })
  );

  it('Check snapshot', async () => {
    const {asFragment } = render(
      <PusherContext.Provider value={[null, vi.fn]}>
        <Calendar/>
      </PusherContext.Provider>
    )

    expect(await screen.findByText("lundi")).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
