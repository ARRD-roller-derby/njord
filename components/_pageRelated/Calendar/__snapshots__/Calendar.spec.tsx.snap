// Vitest Snapshot v1

exports[`<Calendar /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_lmcue_1"
    data-color-mode="dark"
  >
    <header
      class="_header_lmcue_18"
    >
      <header
        class="_header_1yhnh_1"
      >
        <div
          class="_appName_1yhnh_7"
        >
          <a
            class="_title_1045q_1"
            title="retour à l'accueil"
          >
            Njörd
          </a>
        </div>
        <div
          class="_appName_1yhnh_7"
        >
          <div
            class="_count_snjyq_1"
          >
            -- Dr.
          </div>
          <a
            class="_container_1ssp7_1"
            data-active="false"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="clôche"
                class="_bell_1ssp7_11"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
          <div
            class="_container_1cugo_1"
          >
            <div
              class="_avatar_1cugo_12"
              data-ignore="true"
              data-testid="avatar"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="profil"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </header>
      <div
        class="Toastify"
      />
    </header>
    <div
      class="_menu_lmcue_22"
    >
      <nav
        class="_container_vaf4v_1"
      >
        <a
          class="_link_vaf4v_12"
          data-current="true"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Accueil"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div>
            Accueil
          </div>
        </a>
        <a
          class="_link_vaf4v_12"
          data-current="false"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Calendrier"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div>
            Calendrier
          </div>
        </a>
        <a
          class="_link_vaf4v_12"
          data-current="false"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Ma league"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div>
            Ma league
          </div>
        </a>
        <a
          class="_link_vaf4v_12"
          data-current="false"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Mes équipes"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div>
            Mes équipes
          </div>
        </a>
        <a
          class="_link_vaf4v_12"
          data-current="false"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Mes adresses"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div>
            Mes adresses
          </div>
        </a>
        <a
          class="_link_vaf4v_12"
          data-current="false"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2715%27%20height=%2715%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Mon inventaire"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <div>
            Mon inventaire
          </div>
        </a>
      </nav>
    </div>
    <main
      class="_main_lmcue_12"
    >
      <div
        class="_box_lmcue_31"
      >
        <div
          class="_container_1eub2_1"
        >
          <h1
            class="_title_1eub2_9"
          >
            août 2022
          </h1>
          <div
            class="_calendar_1eub2_13"
            data-ismobile="false"
          >
            <div
              class="_header_1eub2_19"
            >
              lundi
            </div>
            <div
              class="_header_1eub2_19"
            >
              mardi
            </div>
            <div
              class="_header_1eub2_19"
            >
              mercredi
            </div>
            <div
              class="_header_1eub2_19"
            >
              jeudi
            </div>
            <div
              class="_header_1eub2_19"
            >
              vendredi
            </div>
            <div
              class="_header_1eub2_19"
            >
              samedi
            </div>
            <div
              class="_header_1eub2_19"
            >
              dimanche
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                1
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                2
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                3
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                4
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                5
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                6
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="true"
              >
                7
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                8
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                9
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                10
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                11
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                12
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                13
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                14
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                15
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                16
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                17
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                18
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                19
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                20
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                21
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                22
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                23
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                24
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                25
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                26
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                27
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                28
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                29
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                30
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                31
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                1
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                2
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                3
              </div>
            </div>
            <div
              class="_day_cib0g_2"
              data-currentmonth="false"
            >
              <div
                class="_dayNum_cib0g_21"
                data-today="false"
              >
                4
              </div>
            </div>
          </div>
          <div
            class="_buttons_1eub2_29"
          >
            <button>
              précedent
            </button>
            <button>
              suivant
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
